data-loader/core/src/main/java/com/scalar/db/dataloader/core/dataexport/producer/CsvProducerTask.java

```diff
@@ -0,0 +1,154 @@
+   */
+  private String convertResultToCsv(Result result) {
+    // Initialization
+    StringBuilder stringBuilder = new StringBuilder();
+    LinkedHashSet<String> tableColumnNames = tableMetadata.getColumnNames();
+    Iterator<String> iterator = tableColumnNames.iterator();
+
+    try {
+      // Loop over the result data list
+      while (iterator.hasNext()) {
+        String columnName = iterator.next();
+
+        // Skip the field if it can be ignored based on check
+        boolean columnNotProjected = !projectedColumnsSet.contains(columnName);
+        boolean isMetadataColumn =
+            ConsensusCommitUtils.isTransactionMetaColumn(columnName, tableMetadata);
+        if (columnNotProjected || (!includeMetadata && isMetadataColumn)) {
+          continue;
+        }
+
+        // Convert each value to a string value and add to the StringBuilder
+        stringBuilder.append(
+            convertToString(result, columnName, dataTypeByColumnName.get(columnName)));
+
+        if (iterator.hasNext()) {
+          stringBuilder.append(delimiter);
+        }
+      }
+
+      // Double check and remove the character if it's a delimiter. This can occur when the last
+      // added column was not the last iterator field and did get a delimiter
+      CsvUtil.removeTrailingDelimiter(stringBuilder, delimiter);
+
+      stringBuilder.append(System.lineSeparator());
+
+      return stringBuilder.toString();
+    } catch (UnsupportedOperationException e) {
+      logger.error(
+          CoreError.DATA_LOADER_VALUE_TO_STRING_CONVERSION_FAILED.buildMessage(e.getMessage()));
+    }
+    return "";
+  }
+
+  /**
+   * Convert result column value to string
+   *
+   * @param result ScalarDB result
+   * @param columnName column name
+   * @param dataType datatype of the column
+   * @return value of result converted to string
+   */
+  private String convertToString(Result result, String columnName, DataType dataType) {
+    if (result.isNull(columnName)) {
+      return null;
+    }
+    String value = "";
+    switch (dataType) {
+      case INT:
+        value = Integer.toString(result.getInt(columnName));
+        break;
+      case BIGINT:
+        value = Long.toString(result.getBigInt(columnName));
+        break;
+      case FLOAT:
+        value = DecimalUtil.convertToNonScientific(result.getFloat(columnName));
+        break;
+      case DOUBLE:
+        value = DecimalUtil.convertToNonScientific(result.getDouble(columnName));
+        break;
+      case BLOB:
+        byte[] encoded = Base64.getEncoder().encode(result.getBlobAsBytes(columnName));
+        value = new String(encoded, Charset.defaultCharset());
+        break;
+      case BOOLEAN:
+        value = Boolean.toString(result.getBoolean(columnName));
+        break;
+      case TEXT:
+        value = result.getText(columnName);
+        break;
+      default:
+        logger.error(
+            CoreError.DATA_LOADER_CONVERT_TO_STRING_FAILED.buildMessage(dataType.toString()));
```

Comment:
We can throw an `AssertionError` since this code should never be reached.

```suggestion
      default:
        throw new AssertionError("Unkown data type:" + dataType);
```